controller/export/github.go

```diff
@@ -0,0 +1,169 @@
+package export
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/neuvector/neuvector/share"
+	"github.com/neuvector/neuvector/share/utils"
+)
+
+const githubApiVersion = "2022-11-28"
+const githubRepoContentUrl = "https://api.github.com/repos/%s/%s/contents/%s"
+
+type gitHubAPI_Comitter struct {
+	Name  string `json:"name"`
+	Email string `json:"email"`
+}
+
+type gitHubAPI_PutRepoContent_RequestBody struct {
+	Message   string             `json:"message"`
+	Committer gitHubAPI_Comitter `json:"comitter"`
+	Content   string             `json:"content"`
+	SHA       string             `json:"sha"`
+}
+
+type gitHubAPI_GetRepoContent_ResponseBody struct {
+	Sha string `json:"sha"`
+}
+
+type githubRepo struct {
+	owner  string
+	name   string
+	branch string
+}
+
+type githubCommitter struct {
+	name                string
+	email               string
+	personalAccessToken string
+}
+
+type GitHubExport struct {
+	repo          githubRepo
+	committer     githubCommitter
+	commitMessage string
+	filePath      string
+	fileContents  []byte
+	url           *url.URL
+}
+
+func (exp GitHubExport) Do() error {
```

Comment:
What about the proxy support for exporting to github?

Reply:
Let's create a jira case. We can do this as bug fix.